conda_smithy/lint_recipe.py

```diff
@@ -402,6 +385,14 @@ def lintify(meta, recipe_dir=None, conda_forge=False):
             "[here](https://conda-forge.org/docs/maintainer/knowledge_base.html#compilers)."
         )
 
+    # 22:
+    for source_section in sources_section:
+        if "git_url" in source_section or "git_tag" in source_section:
+            lints.append(
+                "Building from git repositories is not allowed; please build from "
+                "a source tarball. [See the docs here.](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)"
+            )
```

Comment:
Actually I'm not sure this is a good idea. AFAIK go packages in particular need to clone from GitHub.

cc @sodre

Reply:
Ahh that is a fair point, if we can confirm.

Reply:
I can confirm that go packages are often built directly from git.